### PR TITLE
convert EmojiIcon component from class based to function based component

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

This PR converts the EmojiIcon component from class based component to functional based component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24753

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by jespino — APPROVED** [review]
It looks good to me. Thanks @tanmaythole! :partying_face:

File: webapp/channels/src/components/widgets/icons/emoji_icon.tsx [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-                </svg>
-            </span>
-        );
-    }
-}
+const EmojiIcon: FC<React.HTMLAttributes<HTMLSpanElement>> = (props) => {
```
Comment (larkox) [inline webapp/channels/src/components/widgets/icons/emoji_icon.tsx:7]: The explicit typing is not needed (i.e. `FC<...>`). You can move the props types inside the parenthesis and should be enough.

**Review by larkox — APPROVED** [review]
LGTM! Just the same minor thing I have flagged in a few of your PRs.

File: webapp/channels/src/components/widgets/icons/emoji_icon.tsx [hunk 2]
```diff
@@ -3,20 +3,22 @@
...
+            </svg>
+        </span>
+    );
+};
+
+export default EmojiIcon;
```
Comment (jespino) [inline webapp/channels/src/components/widgets/icons/emoji_icon.tsx:24]: The component was a `PureComponent` so we need to wrap it in a `React.memo` call here:
```suggestion
export default React.memo(EmojiIcon);
```

**Review by jespino — CHANGES_REQUESTED** [review]
A tiny improvement needed.

**Review by jespino — APPROVED** [review]
No looks good to me. Thanks @tanmaythole! :tada:

Take into account that the `React.memo` change can break the tests.

**Comment by coltoneshaw** [comment]
@lindy65 - I'll take QA here based on [your post](https://community.mattermost.com/core/pl/f8ug44zcft8mbkrda16niscygh).

**Comment by lindy65** [comment]
Thanks @coltoneshaw 👍 

@larkox - please run E2E tests on this PR

**Comment by larkox** [comment]
@lindy65 e2e tests seem to be failing on non-related tests only. This should be good to merge.

**Review by lindy65 — APPROVED** [review]
Thanks @tanmaythole - looks good to me 👍 

Thanks for QA-ing @coltoneshaw :)

cc @larkox

**Pull request merged**